Python/pythonrun.c

```diff
@@ -1199,6 +1199,36 @@ print_exception_notes(struct exception_print_context *ctx, PyObject *notes)
     return -1;
 }
 
+static int
+get_exception_notes(struct exception_print_context *ctx, PyObject *value, PyObject **notes) {
+    PyObject *note = NULL;
+
+    if (_PyObject_LookupAttr(value, &_Py_ID(__notes__), notes) < 0) {
+        PyObject *type, *errvalue, *tback;
+        PyErr_Fetch(&type, &errvalue, &tback);
```

Comment:
Wait, why actually are we doing PyErr_Fetch here? We're supposed to be printing the exception ``value`` which is passed into this function as an arg.

Reply:
When in the test we have this:

```py
        class BrokenException(Exception):
            broken = False
            def __getattr__(self, name):
                if self.broken:
                    raise ValueError(f'no {name}')
                raise AttributeError(name)

        e = BrokenException(123)
        vanilla = self.get_report(e)
```

the `value` is the `BrokenException(123)`, what we are doing here is the equivalent of:

```py
        try:
            self.__notes__ = getattr(exc_value, '__notes__', None)     # _PyObject_LookupAttr
        except Exception as e:                                         # PyErr_Fetch
            self.__notes__ = [
                f'Ignored error getting __notes__: {_safe_string(e, "__notes__", repr)}'] 
```

we are catching the `ValueError(f'no {name}')` raised during the `getattr` to add it's `repr` to the notes.